src/main/java/com/lothrazar/cyclic/block/crafter/TileCrafter.java

```diff
@@ -211,6 +213,35 @@ private boolean hasFreeSpace(IItemHandler inv, ItemStack output) {
     return test.isEmpty(); //empty means all of it was allowed to go in
   }
 
+  // This could be done better, but it works so ¯\_(ツ)_/¯
+  private boolean checkInput(IItemHandler inv) {
+	  IItemHandler gridHandler = this.gridCap.orElse(null);
+	  List<ItemStack> inputStacks = new ArrayList<ItemStack>();
+	  List<ItemStack> gridStacks = new ArrayList<ItemStack>();
+	  for(int i = 0; i < inv.getSlots(); i++) {
+		  inputStacks.add(inv.getStackInSlot(i).copy());
+	  }
+	  for(int i = 0; i < gridHandler.getSlots(); i++) {
+		  gridStacks.add(gridHandler.getStackInSlot(i).copy());
+	  }
+	  for(ItemStack stack : inputStacks) {
+		  List<ItemStack> lolbit = new ArrayList<ItemStack>();
```

Comment:
aha so lolbit are the matches so we know what to clear out when we are removingall i got it. works in my testing 👍